lib/Mojo/Util.pm

```diff
@@ -183,6 +184,30 @@ sub monkey_patch {
   *{"${class}::$_"} = set_subname("${class}::$_", $patch{$_}) for keys %patch;
 }
 
+sub network_contains {
+  my ($cidr, $addr) = @_;
+  return undef unless length $cidr && length $addr;
+
+  # Parse inputs
+  my ($net, $mask) = split m!/!, $cidr, 2;
+  my $v6 = $net =~ /:/;
+  return undef if $v6 xor $addr =~ /:/;
+
+  # Convert addresses to binary
+  return undef unless $net  = $v6 ? inet_pton(AF_INET6, $net)  : inet_aton($net);
+  return undef unless $addr = $v6 ? inet_pton(AF_INET6, $addr) : inet_aton($addr);
+  my $length = $v6 ? 128 : 32;
+
+  # Apply mask if given
+  if (defined $mask) {
+    $mask = pack "B$length", '1' x $mask;
+    $addr &= $mask;
+  }
```

Comment:
Can this be golfed into `$addr &= ($mask = pack "B$length", '1' x $mask) if defined $mask`?

Reply:
wow

Reply:
Actually I don't even need to keep the updated `$mask` variable so it doesn't need the extra complexity of the assignment inside the parens